mitxgraders/baseclasses.py

```diff
@@ -205,6 +207,60 @@ def log_output(self):
         content = "\n".join(self.debuglog)
         return "<pre>{content}</pre>".format(content=content)
 
+    @classmethod
+    def ensure_text_inputs(cls, student_input, allow_lists=True, allow_single=True):
+        """
+        Ensures that student_input is a list of text strings.
+
+        NOTE: Overriden by ItemGrader to ensure input is a single text string.
+        """
+        list_errmsg = None
+        single_errmsg = None
+
+        if allow_lists and isinstance(student_input, list):
+            try:
+                return Schema([text_string])(student_input)
+            except MultipleInvalid as error:
+                pos = error.path[0]
+                list_errmsg = (
+                    "Expected a list of text strings for student_input, but "
+                    "item at position {pos} has {thetype}"
+                    .format(pos=pos, thetype=type(student_input[pos])))
+
+        if allow_lists and not isinstance(student_input, list):
+            list_errmsg = (
+                "Expected student_input to be a list of text strings, but "
+                "received {}".format(type(student_input)))
+
+        if allow_single:
+            try:
+                return Schema(text_string)(student_input)
+            except MultipleInvalid as error:
+                single_errmsg = (
+                    "Expected string for student_input, received {}"
+                    .format(type(student_input)))
+
+        if allow_lists and allow_single:
+            msg = ("The student_input passed to a grader should be:\n"
+                   " - a text string for problems with a single input box\n"
+                   " - a list of text strings for problems with multiple input boxes\n"
+                   "Received student_input of {}").format(type(student_input))
+            raise ConfigError(msg)
+        elif allow_lists:
+            raise ConfigError(list_errmsg)
+        elif allow_single:
+            raise ConfigError(single_errmsg)
+        else:
+            raise ValueError('at least one of (allow_lists, allow_single) must be True.')
+
+        try:
```

Comment:
It seems like you'll never get to here. Why is this code present?

Reply:
It currently cannot be reached by using graders. It could be reached by a developer in the future who does something exceedingly silly:

```python
from mitxgraders.baseclasses import AbstractGrader

AbstractGrader.ensure_text_inputs('cat', allow_lists=False, allow_single=False)
# raises ValueError: At least one of (allow_lists, allow_single) must be True.
```

This is one of the lines that lacks coverage but is easy to test.

Reply:
But everything should be caught by the `else` statement above...